lib/application.js

```diff
@@ -202,23 +202,42 @@ class Application extends api.events.EventEmitter {
     });
   }
 
+  loadSchemas(callback) {
+    const dir = api.path.join(this.dir, 'schemas');
+    const loaderPath = api.path.join(dir, 'config.js');
+    impress.createScript(this, loaderPath, (err, getMsConfig) => {
+      if (err) {
+        this.log.error(err);
+        callback(err);
+        return;
+      }
+
+      const { options, config } = getMsConfig();
+      options.context = this.sandbox;
+
+      api.metaschema.fs
+        .load(dir, options, config)
+        .then(([errors, schema]) => {
+          if (errors.length > 0) {
+            this.log.error(errors);
+            callback(errors);
+            return;
+          }
+          this.schema = schema;
+          callback(null);
+        }, callback)
+        .catch(callback);
```

Comment:
It is now possible that the callback will be called more than once, and if we need to keep using promises here, I think the better way of doing this would be:
```js
      api.metaschema.fs
        .load(dir, options, config)
        .then(([errors, schema]) => {
          if (errors.length > 0) {
            this.log.error(errors);
            return Promise.reject(errors);
          }
          this.schema = schema;
        })
        .then(callback, callback);
```
But in my opinion, an even better solution would be to just use `util.callbackify()` to avoid mixing two styles:
```js
      const loadSchema = api.util.callbackify(api.metaschema.fs.load);
      loadSchema(dir, options, config, (err, result) => {
        if (err) {
          callback(err);
          return;
        }
        const [errors, schema] = result;
        if (errors.length > 0) {
          this.log.error(errors);
          callback(errors);
          return;
        }
        this.schema = schema;
        callback(null);
      }
```